sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1328,6 +1332,155 @@ void captureSessionToken(RxDocumentServiceRequest request, RxDocumentServiceResp
         return queryDocuments(collectionLink, "SELECT * FROM r", options);
     }
 
+    @Override
+    public Mono<FeedResponse<Document>> readMany(
+        List<Pair<String, PartitionKey>> itemKeyList,
+        String collectionLink) {
+        
+        DocumentCollection collection = 
+            this.readCollection(collectionLink, new RequestOptions()).block().getResource();
+
+        return this.readCollection(collectionLink, new RequestOptions())
+            .flatMap(documentCollectionResourceResponse -> {
+                         Map<PartitionKeyRange, List<Pair<String, PartitionKey>>> partitionRangeItemKeyMap =
+                             new HashMap<>();
```

Comment:
shouldn't this be concurrentHashMap as it is modified on a flatMap?

Reply:
Did minor refactoring to ensure that these calls are only made once in the flatmap and no other thread is going to write on to this. Also made the rangeQueryMap unmodifiable